pkg/controllers/interruption/controller.go

```diff
@@ -155,6 +158,10 @@ func (c *Controller) handleMessage(ctx context.Context, machineInstanceIDMap map
 			continue
 		}
 		node := nodeInstanceIDMap[instanceID]
+		receivedMessages.With(prometheus.Labels{
```

Comment:
What happens if the message has multiple instance IDs that are part of different provisioners?

Reply:
Wouldn't that be handled from this being in the loop of all the instance ID's in the message, and looking up the provisioner by using the label on the machine directly for each instance ID?